python/metatensor-torch/tests/learn/module_map.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+import torch
+from torch.nn import Module, Sigmoid
+
+from metatensor.torch import allclose_raise
+from metatensor.torch.learn.nn import Linear, ModuleMap
+
+from .utils import TORCH_KWARGS, random_single_block_no_components_tensor_map
+
+
+class MockModule(Module):
+    def __init__(self, in_features, out_features):
+        super().__init__()
+        self._linear = torch.nn.Linear(in_features, out_features)
+        self._activation = Sigmoid()
+        self._last_layer = torch.nn.Linear(out_features, 1)
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        return self._last_layer(self._activation(self._linear(input)))
+
+
+class TestModuleMap:
+    @pytest.fixture(autouse=True)
+    def set_random_generator(self):
+        """Set the random generator to same seed before each test is run.
+        Otherwise test behaviour is dependend on the order of the tests
+        in this file and the number of parameters of the test.
+        """
+        torch.random.manual_seed(122578741812)
+        torch.set_default_device(TORCH_KWARGS["device"])
+        torch.set_default_dtype(TORCH_KWARGS["dtype"])
+
+    @pytest.mark.parametrize(
+        "tensor",
+        [
+            random_single_block_no_components_tensor_map(True, True),
+        ],
+    )
+    def test_module_tensor(self, tensor):
```

Comment:
the parametrize does not seem to be doing anything here. Is this intended to parametrize the `use_metatensor_torch`?

```suggestion
    @pytest.mark.parametrize("use_metatensor_torch", (True, False))
    def test_module_tensor(self, use_metatensor_torch):
    use_torch=True
    tensor = random_single_block_no_components_tensor_map(use_torch, use_metatensor_torch)
```

Otherwise I would just remove the whole parametrize call (here and in the other tests)

Reply:
Yes there was the ambition in adding more tensor maps with multiple blocks, but I think this should be actually a separate test. I'll remove the parameterize

Reply:
I use a fixture from utils (with an explicit import!) and my plan is to move `random_single_block_no_components_tensor_map` in another PR to `metatensor.operations.test_utils.random_tensor` as discussed in https://github.com/lab-cosmo/metatensor/issues/451